docs/examples/my-fuel-dapp/frontend/src/App.tsx

```diff
@@ -24,30 +24,45 @@ const contract = ContractAbi__factory.connect(CONTRACT_ID, wallet);
 
 function App() {
   const [counter, setCounter] = useState(0);
+  const [balance, setBalance] = useState(0); 
 
   useEffect(() => {
     async function main() {
       // Executes the counter function to query the current contract state
       // the `.get()` is read-only, because of this it don't expand coins.
-      const { value } = await contract.functions.counter().get();
-      setCounter(Number(value));
+        const { value } = await contract.functions.counter().get();
+        setCounter(Number(value));
+        setBalance(Number(value));
     }
     main();
   }, []);
 
+  useEffect(() => {
+    let active = true;
+    async function fetch(){
+      if (active) {
+        // Creates a transactions to call the increment function passing the amount
+        // we want to increment, because it creates a TX and updates the contract state
+        // this requires the wallet to have enough coins to cover the costs and also
+        // to sign the Transaction
+        const { value } = await contract.functions.increment(1).call();
+        setBalance(Number(value));
+      }
+    }
+    fetch();
+    return () => {
```

Comment:
I think it would be great to add a loading state to let the user know tx has been executed instead of just blocking the button. And change button text to `Incrementing...`

Reply:
I do agree with you. But, the blocking only occurs if the user clicks the button repeatedly at a very fast speed, so it's clearly not even noticeable. To make a loading state here would add some "latency" to the operation that is meant to be quick. But I will try doing it and see how the experience feels.

Reply:
I've put the loading state, but the `Incrementing...` looked **very weird** because the operation is almost instant. So what I did was updating disabled attribute on the button according to loading state and it feels better now.